agent/fw_drupal8.c

```diff
@@ -534,19 +534,29 @@ NR_PHP_WRAPPER(nr_drupal8_name_the_wt_via_symfony) {
 NR_PHP_WRAPPER_END
 
 void nr_drupal_version() {
-  char* string = "Drupal::VERSION;";
-  zval retval;
-  int result
-      = zend_eval_string(string, &retval, "Retrieve Drupal Version");
-  
+  zval* zval_version = NULL;
+  zend_class_entry* class_entry = NULL;
+
+  class_entry = nr_php_find_class("drupal");
+  if (NULL == class_entry) {
+    nrl_verbosedebug(NRL_INSTRUMENT, "%s: 'Drupal' class not found", __func__);
+    return;
+  }
+
+  zval_version = nr_php_get_class_constant(class_entry, "VERSION");
+  if (NULL == zval_version) {
+    nrl_verbosedebug(NRL_INSTRUMENT, "%s: Drupal does not have VERSION",
+                     __func__);
+    return;
+  }
+
   // Add php package to transaction
-  if (result == SUCCESS) {
-    if (Z_TYPE(retval) == IS_STRING) {
-      char* version = Z_STRVAL(retval);
-      nr_txn_add_php_package(NRPRG(txn), "drupal/core", version);
-    }
-    zval_dtor(&retval);
+  if (IS_STRING == Z_TYPE_P(zval_version)) {
```

Comment:
Consider using a robust `nr_php_is_zval_valid_string` to check if `zval_version` is valid:
```suggestion
  if (nr_php_is_zval_valid_string(zval_version)) {
```